src/test/scala/chiselTests/AsyncResetSpec.scala

```diff
@@ -191,6 +191,54 @@ class AsyncResetSpec extends ChiselFlatSpec {
     assertTesterPasses(new AsyncResetQueueTester)
   }
 
+  it should "support SInt regs" in {
+    assertTesterPasses(new BasicTester {
+      // Also check that it traces through wires
+      val initValue = Wire(SInt())
+      val reg = withReset(reset.asAsyncReset)(RegNext(initValue, 27.S))
```

Comment:
How does this specifically test asnyc reset? Is `reset` still sync reset?

Reply:
These tests aren't specifically testing that it's actually async reset, they assume that is tested by other tests (which it is, see `AsyncResetTester` and others above). They are just checking that non-UInt types are supported since they were tripping FIRRTL previously due to Chisel emitting SInt literals as `asSInt(UInt(...))` (and similar for the others).